### PR TITLE
Add a step in git actions CI to test make check when LLVM is set to none

### PR DESCRIPTION
**Pull request by bhavanijayakumaran**
To address https://github.com/Cray/chapel-private/issues/4220
Signed-off-by: bhavanijayakumaran <82669529+bhavanijayakumaran@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -84,6 +84,21 @@ jobs:
...
+      image: ghcr.io/${{ github.repository_owner }}/chapel-github-ci:latest
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.github_token }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: make check_llvm_none
+      run: |
+       CHPL_LLVM=none ./util/printchplenv --all make make check
+       CHPL_HOME=$PWD ./util/test/run-in-test-venv.bash ./examples/hello.chpl
```
Comment (riftEmber) [inline .github/workflows/CI.yml:99]: I don't think it hurts to have this, but is there a specific reason we test `hello.chpl` in the `CHPL_LLVM=none` configuration but not the usual make check configuration?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -84,6 +84,21 @@ jobs:
       run: |
         CHPL_HOME=$PWD make run-dyno-linters
 
+  make_check_llvm_none:
```
Comment (riftEmber) [inline .github/workflows/CI.yml:87]: It would be nice to declare this right next to the usual `make_check` job, since there is a lot of duplicate code between the two we don't want to forget to update together.

**Review by riftEmber — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -84,6 +84,21 @@ jobs:
...
+    container:
+      image: ghcr.io/${{ github.repository_owner }}/chapel-github-ci:latest
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.github_token }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: make check_llvm_none
+      run: |
+       CHPL_LLVM=none ./util/printchplenv --all make make check
```
Comment (riftEmber) [inline .github/workflows/CI.yml:98]: I looked closer at the non-llvm-none version of this job and realized this is not running the same command -- I assume it is meant to be

**Review by riftEmber — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml (hunk 3 above)
Comment (bhavanijayakumaran) [reply .github/workflows/CI.yml:98]: Hi Anna, This PR is still work in progress.
Just tried to push to to github to understand the behavior.

Comment (riftEmber) [reply .github/workflows/CI.yml:98]: Oops, sorry

**Comment by bhavanijayakumaran** [comment]
This is opened just for testing.
